R/format_GLA.R

```diff
@@ -0,0 +1,688 @@
+                                                                     TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$LayingComplete))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  ObservedHatch = suppressWarnings(dplyr::case_when(grepl("/", .data$ObservedHatch) ~ lubridate::dmy(.data$ObservedHatch, quiet = TRUE),
+                                                                    TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$ObservedHatch))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  Hatchlings = as.integer(.data$Hatchlings),
+                  Fledglings = as.integer(.data$Fledglings),
+                  MaleRing = as.character(.data$MaleRing),
+                  FemaleRing = as.character(.data$FemaleRing),
+                  ClutchSize = as.integer(.data$ClutchSize)) %>%
+
+    ## Arrange
+    dplyr::arrange(.data$PopID, .data$BreedingSeason, .data$LocationID, .data$FirstEggDate) %>%
+
+    ## Create additional variables that will be used in multiple data tables
+    dplyr::mutate(ClutchSize_max = dplyr::case_when(.data$ClutchComplete == "TRUE" ~ as.numeric(.data$ClutchSize),
+                                                    .data$ClutchComplete ==  "FALSE" ~ Inf),
+                  Species = dplyr::case_when(.data$Species == "greti" ~ species_codes[species_codes$SpeciesID == 14640,]$Species,
+                                             .data$Species == "bluti" ~ species_codes[species_codes$SpeciesID == 14620,]$Species,
+                                             .data$Species == "piefl" ~ species_codes[species_codes$SpeciesID == 13490,]$Species,
+                                             .data$Species == "nutha" ~ species_codes[species_codes$SpeciesID == 14790,]$Species,
+                                             .data$Species == "tresp" ~ species_codes[species_codes$SpeciesID == 15980,]$Species),
+                  BroodSize_observed = .data$ClutchSize - as.integer(.data$UnhatchedEggs),
+                  PopID = dplyr::case_when(.data$PopID == "cashel" ~ "CAS",
+                                           .data$PopID == "garscube" ~ "GAR",
+                                           .data$PopID == "kelvingrove_park" ~ "KEL",
+                                           .data$PopID == "sallochy" ~ "SAL",
+                                           .data$PopID == "SCENE" ~ "SCE",
+                                           TRUE ~ as.character(.data$PopID)),
+                  LayDate_observed = .data$FirstEggDate) %>%
+
+    ## Rename
+    dplyr::rename(FemaleID = .data$FemaleRing,
+                  MaleID = .data$MaleRing,
+                  LayDate_min = .data$FirstEggDate,
+                  LayDate_max = .data$LayingComplete,
```

Comment:
`LayDate` should be the date the first egg was laid, so I don't think we should use `LayingComplete` here. If there is no info on LayDate uncertainty, maybe it's better to just keep these as NA.